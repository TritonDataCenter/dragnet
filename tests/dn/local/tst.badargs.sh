#!/bin/bash
#
# tst.badargs.sh: test miscellaneous bad arguments
#

. $(dirname $0)/../common.sh

set -o pipefail

file=$DN_DATADIR/2014/05-01/one.log
function try
{
	if dn scan "$@" input 2>&1 | head -2; then
		echo "unexpected success (args: $@)"
		exit 1
	fi

	return 0
}

dn_clear_config
dn datasource-list
dn datasource-list -v
dn datasource-add --path=$file input
dn datasource-list
dn datasource-list -v
dn datasource-show input
dn datasource-show -v input

try -b host -b req.method,x[=bar]
try -b host -b req.method,[]
try -b host -b req.method,foo[
try -f '{'
try -f '{ "junk": [ "foo", "bar" ] }'

dn datasource-remove input
dn datasource-add --path=$file --data-format=junk input
try
dn_clear_config
