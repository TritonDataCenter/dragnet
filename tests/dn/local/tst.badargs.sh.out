dn: bad value for "breakdowns" ("req.method,x[=bar]"): missing attribute name
usage: dn SUBCOMMAND [OPTIONS] ARGS
dn: bad value for "breakdowns" ("req.method,[]"): missing field name
usage: dn SUBCOMMAND [OPTIONS] ARGS
dn: bad value for "breakdowns" ("req.method,foo["): unexpected end of string
usage: dn SUBCOMMAND [OPTIONS] ARGS
dn: invalid filter: Unexpected end of input
usage: dn SUBCOMMAND [OPTIONS] ARGS
dn: invalid query: invalid filter: predicate { junk: [ 'foo', 'bar' ] }: unknown operator "junk"
dn: unsupported format: "junk"
